lightningd/plugin.c

```diff
@@ -672,6 +672,7 @@ static bool plugin_rpcmethod_add(struct plugin *plugin,
 	cmd = notleak(tal(plugin, struct json_command));
 	cmd->name = json_strdup(cmd, buffer, nametok);
 	cmd->description = json_strdup(cmd, buffer, desctok);
+	cmd->category = CMD_PLUGIN;
```

Comment:
Should not plugins indicate the category of their commands directly, via some optional interface?

Reply:
I think it is useful to tag plugin commands differently than native commands. However we can always do both.

Reply:
I suggest that plugins should be "seamless".  We are moving functionality from "core" of lightningd to plugins, see `pay` and `autocleaninvoice`.

Reply:
I agree for not externally installed plugins like `pay` and `waitinvoice`, but I suggest that externally installed plugins (i.e. most of the time little Python scripts) stay tagged as "add-ons".

Reply:
That's why you should be able to indicate the command category using some optional itnerface, not just mark a command as "plugin" just because it comes from a plugin.

Reply:
@ZmnSCPxj I think plugin command categories can easily be done in an separate/additional PR